src/squidpy/tl/_sliding_window.py

```diff
@@ -0,0 +1,129 @@
+            Only cells with the same library_key in `adata.obs` are assigned to the same window
+        coord_columns: Tuple[str, str]
+            Tuple of column names in `adata.obs` that specify the coordinates (x, y), e.i. ('globalX', 'globalY')
+        sliding_window_key: str
+            Base name for sliding window columns.
+        overlap: int
+            Overlap size between consecutive windows.
+        %(spatial_key)s
+        copy: bool
+            Whether to return a copy of the AnnData object.
+
+        Returns
+        -------
+        If ``copy = True``, returns the design_matrix
+        Otherwise, stores design_matrix in .obs
+    """
+    start = logg.info(f"Creating {sliding_window_key}")
+
+    x, y = coord_columns
+
+    if x not in adata.obs or y not in adata.obs:
+        adata.obs[x] = adata.obsm[spatial_key][:, 0]
+        adata.obs[y] = adata.obsm[spatial_key][:, 1]
+
+    if library_key is not None and library_key not in adata.obs:
+        raise ValueError(f"Library key '{library_key}' not found in adata.obs")
+
+    if library_key is None:
+        libraries = [None]
+        lib_mask = np.ones(len(adata.obs), dtype=bool)
+        lib_data = adata.obs
+    else:
+        libraries = adata.obs[library_key].unique()
+
+    # Create a DataFrame to store the sliding window assignments
+    sliding_window_df = pd.DataFrame(index=adata.obs.index)
+
+    if overlap == 0:
+        sliding_window_df[sliding_window_key] = np.nan
+    else:
+        grid_len = math.ceil(window_size / overlap)
+        num_grid_systems = grid_len**2
+        for i in range(num_grid_systems):
+            sliding_window_df[f"{sliding_window_key}_{i+1}"] = np.nan
+
+    for lib in libraries:
+        if library_key is not None:
+            lib_mask = adata.obs[library_key] == lib
+            lib_data = adata.obs[lib_mask]
+
+        min_x, min_y = lib_data[x].min(), lib_data[y].min()
+
+        if overlap == 0:
+            xgrid, ygrid = (lib_data[x] - min_x) // window_size, (lib_data[y] - min_y) // window_size
+            xrows = xgrid.max()
+            sliding_window_df.loc[lib_mask, sliding_window_key] = (
+                str(lib) + "_" + ((ygrid * (xrows + 1)) + xgrid).astype(int).astype(str)
+            )
+        else:
+            for i in range(num_grid_systems):
+                grid_col_name = f"{sliding_window_key}_{i+1}"
+                row, col = (i // grid_len), (i % grid_len)
+                x_grid_start = min_x + (overlap * col)
+                y_grid_start = min_y + (overlap * row)
+                xgrid, ygrid = (lib_data[x] - x_grid_start) // window_size, (lib_data[y] - y_grid_start) // window_size
+                xrows = xgrid.max()
+                sliding_window_df.loc[lib_mask, grid_col_name] = (
+                    str(lib) + "_" + f"{1+i}_" + ((ygrid * (xrows + 1)) + xgrid).astype(int).astype(str)
+                ).astype("category")
+
+    if copy:
+        logg.info("Finish", time=start)
```

Comment:
Did you mean to log some end time here?

Reply:
I saw this was also used in the _var_by_distance.py tool. [https://github.com/scverse/squidpy/blob/main/src/squidpy/tl/_var_by_distance.py
```suggestion
        logg.info("Finish", time=start)
```](https://github.com/scverse/squidpy/blob/e6b5b0ccd2739aa50b55c7628e685a6750249d8f/src/squidpy/tl/_var_by_distance.py#L171)

Reply:
@LLehner is this a typo?

Reply:
Probably a leftover from an early draft of the method. Can be removed.

Reply:
The entire logging or just that you log the start time as the end time?

Reply:
In _var_by_distance.py both logging statements can be removed.

Reply:
I will remove it then